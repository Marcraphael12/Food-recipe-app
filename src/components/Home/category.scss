.categories {
  >.category {
    &:nth-child(1) {
      background-color: #002364;
    }

    &:nth-child(2) {
      background-color: #003088;
    }

    &:nth-child(4) {
      background-color: #002364;
    }

    &:nth-child(3) {
      background-color: #003088;
    }

    &:nth-child(5) {
      background-color: #002364;
    }

    &:nth-child(6) {
      background-color: #003088;
    }

    &:nth-child(8) {
      background-color: #002364;
    }

    &:nth-child(7) {
      background-color: #003088;
    }

    &:nth-child(9) {
      background-color: #002364;
    }

    &:nth-child(10) {
      background-color: #003088;
    }

    &:nth-child(12) {
      background-color: #002364;
    }

    &:nth-child(11) {
      background-color: #003088;
    }

    &:nth-child(13) {
      background-color: #002364;
    }

    &:nth-child(14) {
      background-color: #003088;
    }
  }
}
