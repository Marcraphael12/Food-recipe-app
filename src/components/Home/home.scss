.categories {
  padding: 10px;
  width: 100%;
  height: auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  @media only screen and (max-width: 600px) {
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;
    padding: 10px 5px;
  }

  >.category {
    width: 100%;
    max-width: 300px;
    height: auto;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
    padding: 10px;
    margin: 10px;
    border: 2px solid #0d43aa;
    border-radius: 15px;

    @media only screen and (max-width: 700px) {
      width: 50%;
      margin: 0;
      border-radius: 0;
    }

    >img {
      width: 100%;
      height: auto;
    }

    >h2 {
      font-family: "Montserrat", sans-serif;
      font-size: 16px;
      color: #fff;
      padding: 10px 0;
      align-self: flex-end;
    }

    // >p {
    //   width: 100%;
    //   height: auto;

    //   display: flex;
    //   justify-content: center;
    //   align-items: center;

    //   padding: 10px 0;

    //   text-align: center;
    //   font-size: 14px;
    //   color: #000;
    // }

    >a {
      width: 50px;
      height: 50px;
      background: none;
      align-self: flex-end;
      border-radius: 50%;

      >button {
        width: 100%;
        height: 100%;
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 10px 0;
        border-radius: 50%;
        border: 2px solid #fff;
        background: none;
        font-size: 14px;
        color: #fff;
        font-weight: bold;

        &:hover {
          cursor: pointer;
          background-color: #38b6ff;
          opacity: 0.8;
        }
      }
    }
  }
}
