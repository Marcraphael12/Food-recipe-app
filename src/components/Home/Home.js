import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { NavLink } from 'react-router-dom';

import './home.scss';

import fetchCategories, { joinCategory } from '../../redux/API/features/async_thunk';

export const WorkDone = (dataFetched) => {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.food.food);
  const foods = Object.values(data);
  console.log(foods);

  useEffect(() => {
    dispatch(dataFetched());
  }, []);

  return foods;
};

const Home = () => {
  const dispatch = useDispatch();
  const handleJoin = (e) => {
    dispatch(joinCategory(e.target.id));
  };
  const foods = WorkDone(fetchCategories);
  return (
    <ul className="categories">
      {foods.length <= 1 ? 'Loading...'
        : foods.map((food) => (
          <li key={food.category_id} className="category">
            <img src={food.category_image} alt={food.category_name} />
            <h2>{food.category_name}</h2>
            <NavLink to="seemore"><button id={food.category_id} type="button" onClick={handleJoin}>See More</button></NavLink>
          </li>
        )) }
    </ul>
  );
};

export default Home;
