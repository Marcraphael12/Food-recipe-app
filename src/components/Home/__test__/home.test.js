import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import Home from '../Home';

describe('Display the categories list', () => {
  it('Renders correctly every categories from the API', () => {
    const category = {
      category_id: 2,
      category_name: 'one',
      description: 'a category',
      joined: true,
    };
    const categories = TestRenderer.create(
      <Provider store={store}>
        <Home key={category.category_id} categories={category} />
      </Provider>,
    ).toJSON();
    expect(categories).toMatchSnapshot();
  });
});
