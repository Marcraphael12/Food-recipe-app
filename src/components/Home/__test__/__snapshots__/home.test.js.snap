// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display the categories list Renders correctly every categories from the API 1`] = `
<ul
  className="categories"
>
  Loading...
</ul>
`;
